python/Ganga/GPIDev/Lib/GangaList/GangaList.py

```diff
@@ -8,27 +8,32 @@
 from Ganga.Utility.util import containsGangaObjects
 import copy
 import sys
+from functools import partial
 
 from Ganga.Utility.logging import getLogger
 logger = getLogger(modulename=True)
 
 
-def makeGangaList(_list, mapfunction=None, parent=None, preparable=False):
+def makeGangaList(_list, mapfunction=None, parent=None, preparable=False, extra_args=None):
     """Should be used for makeing full gangalists"""
 
     # work with a simple list always
-    if isType(_list, list):
+    if isinstance(_list, list):
         _list = _list
-    elif isType(_list, GangaList):
+    elif isinstance(_list, GangaList):
         _list = getProxyAttr(_list, '_list')
     else:
         _list = [_list]
 
     if mapfunction is not None:
-        _list = map(mapfunction, _list)
+        if extra_args is None:
+            _list = [mapfunction(l) for l in _list]
+        else:
+            new_mapfunction = partial(mapfunction, extra_args=extra_args)
+            _list = [new_mapfunction(l) for l in _list]
 
     result = GangaList()
-    result._list = [ stripProxy(e) for e in _list ]
+    result._list.extend([stripProxy(l) for l in _list])
```

Comment:
This is intentionally subverting the checks around modifying the `_list` item in the Schema of the GangaList as we're stripping (any) proxy here, we want to make an instance as fast as possible and return it.